libseshat/src/libseshat.c

```diff
@@ -148,6 +148,9 @@ int init_lib_seshat(const char *url)
 
     __socket_handle_ = nn_socket(AF_SP, NN_REQ);
 
+    if( 0 > __socket_handle_ ) {
+        LibSeshatError("nn_socket returned failure (%d): %s\n", __socket_handle_, strerror(errno));
+    }
     assert(__socket_handle_ >= 0);
```

Comment:
Agreed. First idea was to remove the asserts as they do not belong in production code. For now, monitoring and leaving the assert, to understand failure. As part of resolution, assert will eventually be replaced with failure return codes.